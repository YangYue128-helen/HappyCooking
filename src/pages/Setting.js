import React, {Component} from 'react';
import {
    StyleSheet,
    Text,
    View,
    TouchableOpacity, Image,
} from 'react-native';
import firebase from "../Firebase";
import { Actions } from 'react-native-router-flux'


const Setting = () => {
    return (
        <View style = {styles.container}>
            <View style={styles.headImage}>
                <Image style={{width:200, height:200, borderRadius:100}}
                       source={require('../images/Me.png')}
                />
            </View>
            <View style={styles.buttonStyle}>
                <TouchableOpacity style = {styles.button}>
                    <Text style={styles.buttonText}>Switch Account</Text>
                </TouchableOpacity>
                <TouchableOpacity
                    style = {styles.button}
                    onPress = { () => {
                        firebase.auth().signOut()
                        Actions.login()
                    }}>
                    <Text style={styles.buttonText}>Logout</Text>
                </TouchableOpacity>
            </View>
        </View>
    )
}

export default Setting

const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: 'rgba(0,0,0,0.3)',
        alignItems: 'center',
        justifyContent: 'center',
    },

    headImage:{
        flex: 1,
        marginTop: 150,
        justifyContent:'flex-end',
        alignItems:'center',
    },
    button: {
        width: 300,
        backgroundColor: '#00251a',
        borderRadius: 25,
        marginVertical: 20,
        paddingVertical: 12,
    },
    buttonText: {
        fontSize: 16,
        fontWeight: '500',
        color:'#ffffff',
        textAlign: 'center',
    },
    buttonStyle: {
        marginTop:50,
        paddingVertical: 0
    }
});