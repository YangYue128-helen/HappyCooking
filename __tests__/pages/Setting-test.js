import React from 'react';
import Setting from "../../src/pages/Setting";

import renderer from 'react-test-renderer';

test('renders correctly', () => {
    const tree = renderer.create(<Setting />).toJSON();
    expect(tree).toMatchSnapshot();
});