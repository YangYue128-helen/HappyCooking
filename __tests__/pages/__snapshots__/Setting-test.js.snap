// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgba(0,0,0,0.3)",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "flex-end",
        "marginTop": 150,
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "borderRadius": 100,
          "height": 200,
          "width": 200,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "marginTop": 50,
        "paddingVertical": 0,
      }
    }
  >
    <View
      accessible={true}
      focusable={false}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#00251a",
          "borderRadius": 25,
          "marginVertical": 20,
          "opacity": 1,
          "paddingVertical": 12,
          "width": 300,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#ffffff",
            "fontSize": 16,
            "fontWeight": "500",
            "textAlign": "center",
          }
        }
      >
        Switch Account
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#00251a",
          "borderRadius": 25,
          "marginVertical": 20,
          "opacity": 1,
          "paddingVertical": 12,
          "width": 300,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#ffffff",
            "fontSize": 16,
            "fontWeight": "500",
            "textAlign": "center",
          }
        }
      >
        Logout
      </Text>
    </View>
  </View>
</View>
`;
